### DISCUSSION TITLE
Creating New Control: `FinancialCalendar`

### DISCUSSION THREAD

**Discussion by sharpninja** [AvaloniaUI/Avalonia#12466] [General]
## Proposed New Control: `FinancialCalendar`

> What does it do?

- Display transactions or budgetary items by date.
- Allow sorting items by Time, Name, Description, Tag or Amount.
- Allow displaying as a delimited list, Grid or as a Glyph (A Glyph is the first unicode (`char`) character of either the Name, Description, or Tag property of an `IFinancialItem` instance).

> What supporting types will be included

- A common `IFinancialItem` interface with minimum properties to support described functionality.
- An `enum` named `GlyphProperty` with values of `Name`, `Description`, `Tag` which specifies which property to use to supply a single character glyph.  
  - _NOTE: If `Tag` is selected, then Segoe MDL2 Assets or equivalent will be used to source the glyph, otherwise the inherited or specified control font will be used._

> What does it look like?

<img width="354" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/16146732/f20d57a6-d071-43ea-b249-70cad8f936b9">

**Comment by timunie** [comment]
I honestly like the idea and your input is welcome. However this looks like a very specialized control to me which would be better suited inside a custom control library than in the main Avalonia repo. You can explore this sample for ideas how to create such a control: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample

**Reply by sharpninja** [reply]
I'm happy to do as either an intentional addition to Calandar (by adding a new `FinancialMonth` view) or as a separate control library.  The biggest issue is having access to the `internal` objects of the `Avalonia.Base` library which the `Calendar` is dependent upon.  I would have to start from scratch without them.

**edit**

Thinking more on this, I think a new view makes the most since because it avoids duplication of code.  Now, maybe that new view is a `TemplatedDay` view instead of a `FinancialMonth` where a `DaysItemsSource` is added to the control, a Property containing the Date is referenced by name and then the `TreeDataTemplate` is used to reference the child data for the Day object.  Finally, a `Calendar.ItemTemplate` property hosts the `TreeDataTemplate`.  This would probably be the least amount of work to implement, then the days can be anything.

**Reply by timunie** [reply]
You can inherit from Calendar if that helps you.

**Reply by sharpninja** [reply]
Digging into the code, CalendarItem is where all the action would happen.  I think duplicating that (Inheritence won't work as the new object needs to extend `ItemsControl`) and updating the constants to include the new object name.
